src/index.js

```diff
@@ -19,70 +49,102 @@ export default class Rollup extends Plugin {
-          moduleName: options.moduleName
-        }
-      ];
-      delete options.format;
-      delete options.dest;
-      delete options.moduleName;
-      return options;
+      return options.targets.map(target => assign({}, options, target));
     }
+    throw new Error('missing targets or dest in options');
   }
 
-  build() {
-    const options = this._loadOptions();
-
-    if (options.entry && !path.isAbsolute(options.entry)) {
-      // if entry is not absolute, make it absolute relative to the inputPath
-      options.entry = this.inputPaths[0] + '/' + options.entry;
-    }
-
-    return heimdall.node('rollup', () => {
-      return require('rollup').rollup(options)
-        .then(bundle => {
-          this._lastBundle = bundle;
+  _patchOutput(nextTree) {
+    let patch = this._lastTree.calculatePatch(nextTree, isEqual);
+    this._lastTree = nextTree;
+    patch.forEach(([op, path, entry]) => {
+      switch (op) {
+        case 'mkdir':
+          fs.mkdirSync(this.outputPath + '/' + path);
+          break;
+        case 'rmdir':
+          fs.rmdirSync(this.outputPath + '/' + path);
+          break;
+        case 'unlink':
+          fs.unlinkSync(this.outputPath + '/' + path);
+          break;
+        case 'create':
+        case 'change':
+          fs.writeFileSync(this.outputPath + '/' + path, entry.content);
+          break;
+      }
+    });
+  }
 
-          options.targets.forEach(target => {
-            let dest = target.dest;
-            let format = target.format;
-            let output = bundle.generate(target).code;
-            let outputPath = this.outputPath + '/'+ dest;
+  _withInputPath(cb) {
+    const dir = process.cwd();
+    return Promise.resolve().then(() => {
+      process.chdir(this.inputPaths[0]);
+    }).then(cb).finally(() => {
+      process.chdir(dir);
+    });
+  }
 
-            fs.mkdirpSync(path.dirname(outputPath)); // needs dependency stuff :P
+  _buildTargets(bundle, options) {
+    let entries = [];
+    this._targetsFor(options).forEach(options => {
+      let dest = options.dest;
+      let { code, map } = bundle.generate(options);
+      if ( options.sourceMap ) {
+        let url;
+        if (options.sourceMap === 'inline') {
+          url = map.toUrl();
+        } else {
+          url = path.basename(dest) + '.map';
+          entries.push(new Entry(dest + '.map', map.toString()));
+        }
+        code += `\n//# sourceMappingURL=${url}\n`;
+      }
+      entries.push(new Entry(dest, code));
+    });
+    return FSTree.fromEntries(entries, {
+      sortAndExpand: true
+    });
+  }
 
-            this.writeFileIfContentChanged(outputPath, output);
+  build() {
+    const options = this._loadOptions();
+    return heimdall.node('rollup', () => {
+      return this._withInputPath(() => {
+        return require('rollup').rollup(options)
+          .then(bundle => {
+            this._lastBundle = bundle;
+            let nextTree = this._buildTargets(bundle, options);
```

Comment:
if there is a rejection, we might be left in a bad state. We should likely reset the cache and trees